app/concepts/book/query/index.rb

```diff
@@ -0,0 +1,53 @@
+module Book::Query
+  class Index
+    SORTINGS = %w[newest popular high_price low_price title_ascending title_descending].freeze
+
+    # rubocop: disable Metrics/AbcSize, CyclomaticComplexity
+    def call(params)
+      @params = params
+
+      case @params[:sort_by]
+      when I18n.t('sortings.system.newest')           then newest(from_category)
+      when I18n.t('sortings.system.popular')          then popular(from_category)
+      when I18n.t('sortings.system.low_price')        then low_price(from_category)
+      when I18n.t('sortings.system.high_price')       then high_price(from_category)
+      when I18n.t('sortings.system.title_ascending')  then title_ascending(from_category)
+      when I18n.t('sortings.system.title_descending') then title_descending(from_category)
+      else title_ascending(from_category)
+      end
+    end
+    # rubocop: enable Metrics/AbcSize, CyclomaticComplexity
+
+    private
+
+    def from_category
+      category_books = (@params[:category_id] ? Category.find(@params[:category_id]).books : Book.all)
```

Comment:
you pass from_category to every method
Le'ts add `initialize`, where @books (Category.find(@params[:category_id]).books : Book.all) will be defined
What if category_id is invalid? Maybe we should do .find_by(id: id) instead of find?

Reply:
`Find` will throw RecordNotFound exception which will be rescuied with 404 page.

Reply:
ok